### PR TITLE
Add libcuba

### PR DESCRIPTION
**Pull request by jschueller**
I'm using git for now instead of the tarball because several patches are required (but I notified the author)

/cc @conda-forge/help-c-cpp

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [ ] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by jschueller** [comment]
ping @conda-forge/help-c-cpp

File: recipes/libcuba/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  name: libcuba
+  version: {{ version }}
+
+source:
+  # url: https://feynarts.de/cuba/Cuba-{{ version }}.tar.gz
+  # sha256: 8d9f532fd2b9561da2272c156ef7be5f3960953e4519c638759f1b52fe03ed52
+
+  # upstream did not respond yet to submitted patches
+  git_rev: "main"
+  git_url: "https://github.com/jschueller/cuba.git"
```
Comment (carterbox) [inline recipes/libcuba/meta.yaml:13]: Do not use a tag from your own fork. There is no easy way for someone to audit how your fork has diverged from the canonical repository.

It's inconvenient for large patches, but text formatted patch included with this recipe is best.

**Review by carterbox — CHANGES_REQUESTED** [review]

File: recipes/libcuba/meta.yaml (hunk 1 above)
Comment (jschueller) [reply recipes/libcuba/meta.yaml]: ok, done

**Comment by jschueller** [comment]
maybe one of @xhochy or @marcelotrevisani  could review it, please ?

**Pull request merged**